types/v56/types.go

```diff
@@ -2671,3 +2692,52 @@ type AdminCatalogRecord struct {
 	Link                    *Link     `xml:"Link,omitempty"`
 	Vdc                     *Metadata `xml:"Metadata,omitempty"`
 }
+
+// EdgeGatewayVnics is a data structure holding information of vNic configuration in NSX-V edge
+// gateway
+type EdgeGatewayVnics struct {
+	XMLName xml.Name `xml:"vnics"`
```

Comment:
Are sure that all fields can't be empty?

Reply:
I only use a couple of them here in "read only" fashion. It never failed but I added tags to majority.

Reply:
In general we match API description, which are mandatory, which aren't. It isn't decision from our point of view.

Reply:
Yes, but this one is used for read-only purposes and I care only about a few fields. I very much doubt we will ever need a "POST" capability. This is edge gateway network card configuration.

Reply:
I find full types.go of previously inserted types were was used minimally which are wrongly matched with required/optional and  as result create unexpected behaviour when you start to use them. Just a tip to avoid that